indexer/src/fastnear_indexer.rs

```diff
@@ -0,0 +1,195 @@
+                        continue;
+                    }
+
+                    let partial_candidate_data = Self::receipt_filter_map(
+                        receipt_execution_outcome.receipt.receipt,
+                        *rollup_id
+                    );
+
+                    if let (Some(partial_data), Some(tx_hash)) = (partial_candidate_data, receipt_execution_outcome.tx_hash) {
+                        let candidate_data = PartialCandidateDataWithBlockTxHash {
+                            rollup_id: *rollup_id,
+                            payloads: partial_data.payloads,
+                            tx_hash,
+                            block_hash: block.block.header.hash,
+                        };
+                        debug!(target: "fastnear_indexer", "Sending candidate data for rollup_id: {}", rollup_id);
+                        Self::send(&candidate_data, publish_sender).await?;
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn stream_latest_blocks(&self) -> mpsc::Receiver<BlockWithTxHashes> {
+        info!(target: "fastnear_indexer", "Starting block stream");
+        let (block_sender, block_receiver) = mpsc::channel(100);
+        let client = self.client.clone();
+
+        tokio::spawn(async move {
+            loop {
+                match Self::fetch_latest_block(&client).await {
+                    Ok(block) => {
+                        if block_sender.send(block.clone()).await.is_err() {
+                            error!(target: "fastnear_indexer", "Failed to send block to channel");
+                            break;
+                        }
+                        info!(target: "fastnear_indexer", "Successfully fetched and sent latest block with id: {}", block.block.header.height);
+                    }
+                    Err(e) => error!(target: "fastnear_indexer", "Error fetching latest block: {:?}", e),
+                }
+                tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+            }
+        });
+
+        block_receiver
+    }
+
+    async fn fetch_latest_block(client: &Client) -> Result<BlockWithTxHashes, Error> {
+        debug!(target: "fastnear_indexer", "Fetching latest block");
+        let response = client.get(FASTNEAR_ENDPOINT)
+            .send()
+            .await
+            .and_then(|r| r.error_for_status())
+            .map_err(|e| Error::NetworkError(e.to_string()))?;
+
+        response.json::<BlockWithTxHashes>()
+            .await
+            .map_err(|e| Error::DeserializeJsonError(e.to_string()))
+    }
+
+    async fn send(candidate_data: &PartialCandidateDataWithBlockTxHash, sender: &Sender<PublishData>) -> Result<(), Error> {
+        trace!(target: "fastnear_indexer", "Sending candidate data: {:?}", candidate_data);
+        for data in candidate_data.clone().payloads {
```

Comment:
we can loop through a ref this to avoid cloning all data
```suggestion
        for data in &candidate_data.payloads {
```

Reply:
That's doesn't work, `PublishPayload` wants actual object and it can't be moved =/